doc/js-perspective-on-c.md

```diff
@@ -0,0 +1,108 @@
+# A JS Developer's Perspective on `C/C++`
+
+## What JS Developers Take For Granted
+
+JS has some serious luxuries. The runtime is forgiving because it is a single threaded, event driven, garbage collected language. The language is forgiving because it is dynamically typed and interpreted. All of that is a function of lines of code in `node`, `v8`, `libuv` and a host of other factors including good design and decent language semantics.
+
+When building native code, those things need to be done manually. Memory will need to be managed.  Threading context will need to be understood. This is not to say that it is impossible to write good code in `C/C++` but it is a lot more to think about than writing JS.
+
+As an example, async in JS can be confusing for new developers. Some docs and blogs were written very early and use patterns like callbacks. Some are from much later and use async generators. Along with structural changes, there have also been stylistic shifts over time. This same thing happened in `C/C++`.
+
+The differences that are seen across docs/blogs/etc stem from `C` being quite old and lots of patterns changed along the way.  Another piece is that `C` and `C++` are very similar, but they are not the same.  Much like `TS` is an extension of `JS`, `C++` is an extension of `C`.  So like `JS/TS` there are `C` and `C++` ways to do things, and in many contexts both ways are "valid."
+
+## Memory Management
+
+Stack and heap allocation are both similar and different to JS and it is important to understand the nuances because they have greatly influenced this library.  In  particular, focus on how everything gets freed.  It's easy to allocate memory but tracking and cleaning up the allocations is where the challenge lies.  Many of the paradigms in `C/C++` are inherently, or explicitly, implemented for resource management.  Because this library is primarily `c++` we will use [RAII](https://en.cppreference.com/w/cpp/language/raii) and OOP to help us along the way.
```

Comment:
```suggestion
Stack and heap allocation are both similar and different to JS and it is important to understand the nuances because they have greatly influenced this library.  In  particular, focus on how everything gets freed.  It's easy to allocate memory but tracking and cleaning up the allocations is where the challenge lies.  Many of the paradigms in `C/C++` are inherently, or explicitly, implemented for resource management.  Because this library is primarily `C++` we will use [RAII](https://en.cppreference.com/w/cpp/language/raii) and OOP to help us along the way.
```

Reply:
Updated.